crates/rust-analyzer/src/lsp_ext.rs

```diff
@@ -242,3 +242,37 @@ pub struct SnippetTextEdit {
     #[serde(skip_serializing_if = "Option::is_none")]
     pub insert_text_format: Option<lsp_types::InsertTextFormat>,
 }
+
+pub enum HoverRequest {}
+
+impl Request for HoverRequest {
+    type Params = lsp_types::HoverParams;
+    type Result = Option<Hover>;
+    const METHOD: &'static str = "textDocument/hover";
+}
+
+#[derive(Debug, PartialEq, Clone, Deserialize, Serialize)]
+pub struct Hover {
+    pub contents: lsp_types::HoverContents,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub range: Option<Range>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub actions: Option<Vec<CommandLinkGroup>>,
+}
```

Comment:
An alternative to this `Hover` would be to do something like:

```
pub struct HoverExt { // Or maybe you could use Hover
    #[serde(flatten)]
    pub hover: lsp_types::Hover,
    
    #[serde(skip_serializing_if = "Option::is_none")]
    pub actions: Option<Vec<CommandLinkGroup>>,
}
```

That way when the base LSP hover changes we will pick them up.